DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs

```diff
@@ -778,15 +778,16 @@ private void HandleOptionsRequest(string playerName, int options)
             if (color < 0 || color > MPColors.Count)
                 return;
 
-            var disallowedSides = GetDisallowedSides().ToList();
-
             // Disallowed sides from client, maps, or game modes do not take random selectors into account
             // So, we need to insert "false" for each random at the beginning of this list AFTER getting them
             // from client, maps, or game modes.
-            for(int i = 0; i < RandomSelectorCount; i++)
-                disallowedSides.Insert(0, false);
+            var randomDisallowedSides = new List<bool>(RandomSelectorCount);
+            for (int i = 0; i < RandomSelectorCount; i++)
+                randomDisallowedSides.Add(false);
+
+            var disallowedSides = randomDisallowedSides.Concat(GetDisallowedSides()).ToArray();
 
-            if (side > 0 && side <= SideCount && disallowedSides[side])
+            if (0 < side && side < SideCount && disallowedSides[side])
```

Comment:
I don't know if the logic here from YR is correct or not. Please double check the original two issues and determine if it's the correct solution to both YR and non-YR